src/Microsoft.Azure.WebJobs.Extensions.Kafka/Output/KafkaProducerAsyncCollector.cs

```diff
@@ -32,14 +34,60 @@ public Task AddAsync(T item, CancellationToken cancellationToken)
 
             object messageToSend = item;
 
-            if (item.GetType() == typeof(string) || item.GetType() == typeof(byte[]))
+            if (item.GetType() == typeof(string))
+            {
+                messageToSend = convertToKafkaEventData(item);
+            }
+
+            if (item.GetType() == typeof(byte[]))
             {
                 messageToSend = new KafkaEventData<T>(item);
             }
 
             return entity.SendAndCreateEntityIfNotExistsAsync(messageToSend, functionInstanceId, cancellationToken);
         }
 
+        private object convertToKafkaEventData(T item)
```

Comment:
Naming convention. In case of C# method name will be start with upper case.  The other methods are the same.